# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientInfo.h
# Commit: af27682c5e7f
# Full Hash: af27682c5e7f86ff17d6ed152c8010e763870bdc
# Author: ssengupta <ssengupta@mozilla.com>
# Date: 2020-05-13 03:44:05
# Regressor Bug: 1635399
# File Overlap Count: 1
# Description:
#   Bug 1635399 - Function PrincipalInfoToPrincipal now returns Result<nsCOMPtr<nsIPrincipal>, nsresult> r=ckerschb
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D73868
# ==============================================================================

diff -r 86b22b1a9f7e -r af27682c5e7f dom/clients/manager/ClientInfo.h
--- a/dom/clients/manager/ClientInfo.h	Tue May 12 23:55:42 2020 +0300
+++ b/dom/clients/manager/ClientInfo.h	Tue May 12 19:02:05 2020 +0000
@@ -94,9 +94,8 @@
   // Determine if the client is in private browsing mode.
   bool IsPrivateBrowsing() const;
 
-  // Get a main-thread nsIPrincipal for the client.  This may return nullptr
-  // if the PrincipalInfo() fails to deserialize for some reason.
-  nsCOMPtr<nsIPrincipal> GetPrincipal() const;
+  // Get a main-thread nsIPrincipal for the client.
+  Result<nsCOMPtr<nsIPrincipal>, nsresult> GetPrincipal() const;
 
   const Maybe<mozilla::ipc::CSPInfo>& GetCspInfo() const;
   void SetCspInfo(const mozilla::ipc::CSPInfo& aCSPInfo);